### PR TITLE
`azurerm_api_management_identity_provider_aad`, `azurerm_api_management_identity_provider_aadb2c` - support `client_library` property

### LINKED ISSUE #18901: Support for `clientLibrary` attribute in `azurerm_api_management_identity_provider_aad` / `b2c`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure API Management Developer Portal is moving from `ADAL` to `MSAL` between now and 2025.
During this time, there is a new field called `clientLibrary` which can be set to ADAL or MSAL to support the transition.

The proposal is to add this new field to the two currently impacted resources.

### New or Affected Resource(s)/Data Source(s)

azurerm_api_management_identity_provider_aad azurerm_api_management_identity_provider_aadb2c

### Potential Terraform Configuration

```hcl
resource "azurerm_api_management_identity_provider_aad" "example" {
  resource_group_name = azurerm_resource_group.example.name
  api_management_name = azurerm_api_management.example.name
  client_id           = "00000000-0000-0000-0000-000000000000"
  client_secret       = "00000000000000000000000000000000"
  allowed_tenants     = ["00000000-0000-0000-0000-000000000000"]
  client_library      = "MSAL"  // new attribute
}

resource "azurerm_api_management_identity_provider_aadb2c" "example" {
  resource_group_name = azurerm_resource_group.example.name
  api_management_name = azurerm_api_management.example.name
  client_id           = azuread_application.example.application_id
  client_secret       = "P@55w0rD!"
  allowed_tenant      = "myb2ctenant.onmicrosoft.com"
  signin_tenant       = "myb2ctenant.onmicrosoft.com"
  authority           = "myb2ctenant.b2clogin.com"
  signin_policy       = "B2C_1_Login"
  signup_policy       = "B2C_1_Signup"
  client_library      = "MSAL"  // new attribute

  depends_on = [azuread_application_password.example]
}
```

### References

- https://learn.microsoft.com/en-us/azure/api-management/breaking-changes/identity-provider-adal-retirement-sep-2025

### PR DESCRIPTION
**Pull request by sinbai**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

Support `client_library` property for resources `azurerm_api_management_identity_provider_aad` and `azurerm_api_management_identity_provider_aadb2c`.

Swagger: https://github.com/Azure/azure-rest-api-specs/blob/8322ff8ff0b9818c49e2e6ef7d02e3d0b0333839/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/definitions.json#L3149

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```
PASS: TestAccAzureRMApiManagementIdentityProviderAADB2C_clientLibrary (2653.77s)
PASS: TestAccAzureRMApiManagementIdentityProviderAADB2C_basic (1714.90s)
PASS: TestAccAzureRMApiManagementIdentityProviderAADB2C_requiresImport (1900.35s)

```
```
PASS: TestAccApiManagementIdentityProviderAAD_clientLibrary (2021.26s)
PASS: TestAccApiManagementIdentityProviderAAD_basic (2148.77s)
PASS: TestAccApiManagementIdentityProviderAAD_update (1976.15s)
PASS: TestAccApiManagementIdentityProviderAAD_requiresImport (1852.06s)
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_api_management_identity_provider_aad` - support for the `client_library` property [GH-18901]
* `azurerm_api_management_identity_provider_aadb2c` - support for the `client_library` property [GH-18901]

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #18901

### CODE REVIEW

File: internal/services/apimanagement/api_management_identity_provider_aad_resource.go [hunk 1]
```diff
@@ -63,6 +63,13 @@ func resourceApiManagementIdentityProviderAAD() *pluginsdk.Resource {
 					ValidateFunc: validation.IsUUID,
 				},
 			},
+
+			"client_library": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 16),
```
Comment (stephybun) [inline internal/services/apimanagement/api_management_identity_provider_aad_resource.go:70]: Where has this validation constraint come from?

Comment (sinbai) [reply internal/services/apimanagement/api_management_identity_provider_aad_resource.go:70]: This is from [swagger](https://github.com/Azure/azure-rest-api-specs/blob/8322ff8ff0b9818c49e2e6ef7d02e3d0b0333839/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2022-08-01/definitions.json#L3149), details see below.

image

**Comment by sschmitzkbnz** [comment]
It would be amazing if this could get merged soon as I am currently deploying APIMs and not having to use azapi_resource_actions to patch the identity provider would make my life so much easier!

**Review by stephybun — APPROVED** [review]
Thanks @sinbai LGTM 👍

**Pull request merged**